runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpPauseStateSingle.java

```diff
@@ -0,0 +1,72 @@
+package net.runelite.client.plugins.xptracker;
+
+import java.time.Instant;
+import java.util.EnumSet;
+import java.util.Set;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import net.runelite.api.Skill;
+
+
+@RequiredArgsConstructor
+public class XpPauseStateSingle
+{
+	@Getter
+	private final Skill skill;
+	private final Set<XpPauseReason> pauseReasons = EnumSet.noneOf(XpPauseReason.class);
+	@Getter
+	private Instant lastXpInstant;
+
+	public boolean isPaused()
+	{
+		return !pauseReasons.isEmpty();
+	}
+
+	public boolean onLogin()
+	{
+		return pauseReasons.remove(XpPauseReason.PAUSED_LOGOUT);
+	}
+
+	public boolean onLogout()
+	{
+		return pauseReasons.add(XpPauseReason.PAUSED_LOGOUT);
+	}
+
+	public boolean onTimeout()
+	{
+		return pauseReasons.add(XpPauseReason.PAUSED_TIMEOUT);
+	}
+
+	public boolean onManualPause()
+	{
+		return pauseReasons.add(XpPauseReason.PAUSE_MANUAL);
+	}
+
+	public boolean onXpChanged()
```

Comment:
Is onXpChanged and onUnPause really necessary as they are same methods? I would just expose clearAll

Reply:
I've simplified a few things in the parent `XpPauseState` class, this function will have slightly different behavior in the next revision.